test/uk/gov/hmrc/preferenceschangednotifier/repository/PreferencesChangedWorkItemRepositorySpec.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.preferenceschangednotifier.repository
+
+import org.mongodb.scala.bson.ObjectId
+import org.mongodb.scala.model.Filters
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
+import play.api.Configuration
+import play.api.test.Helpers
+import uk.gov.hmrc.mongo.test.{DefaultPlayMongoRepositorySupport, MongoSupport}
+import uk.gov.hmrc.mongo.workitem.{ProcessingStatus, WorkItem}
+import uk.gov.hmrc.preferenceschangednotifier.model.PreferencesChangedRef
+
+import java.time.Instant
+import scala.concurrent.ExecutionContext
+
+class PreferencesChangedWorkItemRepositorySpec
```

Comment:
Is it worth adding some tests for indexes on this repo? Also, any tests to verify the read/write formatters of `PreferencesChangedRef`?

Reply:
Good point @kjainsc , thanks!